Orleans/src/Microsoft.Orleans.Host.Abstractions/AzureStorageProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the azure storage providers DLL
+    public static class AzureStorageProviderExtensions
+    {
+        public static void AddAzureStorage(this IStorageProvidersFactory factory, string providerName, string connectionString)
+        {
+            var concreteFactory = ActivatorUtilities.CreateInstance<AzureStorageProviderFactory>(factory.Silo.ApplicationServices);
```

Comment:
I believe that the AzureStorageProviderFactory might not be required if I use the right overload for ActivatorUtilities.CreateInstance, since the factory is there mainly to forward injected service dependencies and separate them from instance specific data/config